src/uct/api/uct_def.h

```diff
@@ -155,16 +156,38 @@ typedef struct uct_iov {
  * @brief Client-Server private data pack callback arguments field mask.
  *
  * The enumeration allows specifying which fields in
- * @ref uct_cm_ep_priv_data_pack_args are present, for backward compatibility support.
+ * @ref uct_cm_ep_priv_data_pack_args are present, for backward compatibility
+ * support.
  */
 enum uct_cm_ep_priv_data_pack_args_field {
     /** Enables @ref uct_cm_ep_priv_data_pack_args::dev_name
-     *  Indicates that dev_name field in uct_cm_ep_priv_data_pack_args_t is valid.
+     *  Indicates that dev_name field in uct_cm_ep_priv_data_pack_args_t is
+     *  valid.
      */
     UCT_CM_EP_PRIV_DATA_PACK_ARGS_FIELD_DEVICE_NAME = UCS_BIT(0)
 };
 
 
+/**
+ * @ingroup UCT_CLIENT_SERVER
+ * @brief Client-Server resolve callback arguments field mask.
+ *
+ * The enumeration allows specifying which fields in
+ * @ref uct_cm_ep_resolve_args are present, for backward compatibility support.
+ */
+enum uct_cm_ep_resolve_args_field {
+    /**
+     * Indicates that @ref uct_cm_ep_resolve_args::dev_name is valid.
+     */
+    UCT_CM_EP_RESOLVE_ARGS_FIELD_DEV_NAME       = UCS_BIT(0),
+
+    /**
+     * Indicates that @ref uct_cm_ep_resolve_args::user_data is valid.
+     */
+    UCT_CM_EP_RESOLVE_ARGS_FIELD_USER_DATA      = UCS_BIT(1)
```

Comment:
user-defined argument should is a mandatory parameter "void *arg"
see other callbacks as example

Reply:
but `uct_ep_params_t::user_data` is not mandatory, should we provide it as undefined or NULL? My thought is we can indicate this not mandatory arg by field mask in param.

Reply:
our convention in other callbacks is that the user-defined argument is mandatory (even if null)
why need to make it different this time?

Reply:
ok